frappe/core/doctype/file/file.py

```diff
@@ -382,8 +382,13 @@ def validate_duplicate_entry(self):
 			filters = {
 				"content_hash": self.content_hash,
 				"is_private": self.is_private,
-				"name": ("!=", self.name),
 			}
+
+			if self.name:
+				filters.update({"name": ("!=", self.name)})
+			else:
+				filters.update({"file_name": self.file_name})
```

Comment:
Can you explain the rationale for this branch?

I understand only using name if it's present but what is reason for using `file_name` if name isn't present?